soql-server-pg/src/main/scala/com/socrata/pg/query/DataSqlizerQuerier.scala

```diff
@@ -49,9 +50,21 @@ trait DataSqlizerQuerier[CT, CV] extends AbstractRepBasedDataSqlizer[CT, CV] wit
       (cid, rep.fromResultSet _, rep.physColumns.length)
     }.toArray
 
+    /*
+     * We need to set a fetchSize on the statement to limit the amount of data we are buffering in memory at
+     * once.  Unfortunately, this currently prevents parallel query execution in postgres because it uses cursors.
+     * Since the major benefit of parallel queries is for queries that take a lot of rows and reduce them to a small number,
+     * if we can tell from the query that it can't return more than sqlFetchSize rows, we can safely set the fetchsize
+     * to 0 and get the benefits of parallel queries.
+     */
+    val fetchSize = analyses.last.limit match {
+      case Some(n) if (n <= sqlFetchSize) => 0
```

Comment:
Let's say we want to turn off parallel query execution for some kind of emergency, and we do that. It will still be totally fine that we are using fetch size of 0 for limited queries (`n <= sqlFetchSize`)? Or would it be nice to be able to return to our old behavior with a marathon configuration change?